crates/storage/src/tables.rs

```diff
@@ -132,6 +132,31 @@ pub mod merkle {
         fuel_types::BlockHeight,
     };
 
+    /// The key for the corresponding `DenseMerkleMetadata` type.
+    /// The `Latest` variant is used to have the access to the latest dense Merkle tree.
+    #[derive(Default, Debug, Clone, serde::Serialize, serde::Deserialize)]
+    pub enum DenseMetadataKey<PrimaryKey> {
+        /// The primary key of the `DenseMerkleMetadata`.
+        Primary(PrimaryKey),
+        #[default]
+        /// The latest `DenseMerkleMetadata` of the table.
+        Latest,
+    }
```

Comment:
I want to understand this better. If `Latest` gives you the entire latest Merkle tree, the `PrimaryKey` gives you a previous tree? Or a subsection?

Maybe this would be obvious if I understood the MTs terminology better.

Reply:
Yes. The Merkle tree that we are using here is append-only. Because of that, we can only add new stuff at the end, and the `Latest` key allows us to do that. At the same time, appending a new leaf doesn't invalidate all previous Merkle trees, and `PrimaryKey` allows us to get the valid `MerkleTree` at some `Key` that we added before. For example, in the case of the `FuelBlocks`, we can get a valid Merkle tree at height X, X-1, .., X - Y and use it in proofs that some block is part of the Merkle tree at a specific height.